src/main/java/org/jenkinsci/plugins/workflow/cps/CpsFlowExecution.java

```diff
@@ -255,6 +255,18 @@
 
     private final AtomicInteger iota = new AtomicInteger();
 
+    /** Number of node IDs to use in lookup table, 500 covers most common flow graphs  */
+    private static final int ID_LOOKUP_TABLE_SIZE = 500;
+
+    /** Preallocated lookup table for small ID values, used instead of interning for speed & simplicity */
```

Comment:
We're creating a ton of garbage strings that must be immediately GC'd if we do that (unless escape analysis catches them, but that's unreliable) -- in any case, valueOf is more expensive than one would think and so is String.intern( ). 

If it was called once per build instead of hundreds of times, I'd agree with you, or if it added any meaningful level of extra complexity.  As is, there's no point in pessimizing it.